Src/Base/AMReX_MFIter.cpp

```diff
@@ -215,9 +215,13 @@ MFIter::~MFIter ()
 void
 MFIter::Finalize ()
 {
+    // avoid double finalize
     if (finalized) return;
     finalized = true;
 
+    // mark as invalid
+    currentIndex = endIndex;
```

Comment:
It won't hurt, although it's already invalid in normal usage, otherwise the MFIter loop won't stop.

Reply:
At least on our side, that's also the only location we would call this:
https://github.com/AMReX-Codes/pyamrex/blob/82d6260d85d70ecc8f3de679b8ed1fc9d8f53758/src/Base/MultiFab.cpp#L72-L75

I was thinking to add some safety if someone calls `mfi.Finalize()` mid loop - but why would they... maybe to abort a loop in a clean way...

Reply:
They can use `goto`.   That might be the cleanest way.  :smile: